src/System.Text.Json/src/System/Text/Json/Serialization/ArrayBufferWriter.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization
+{
+    // Note: this is currently an internal class that will be replaced with a shared version.
```

Comment:
My understanding from the API review is that the shared version will actually be something like a ReadOnlySequenceWriter.  Will that work for the serializer's needs?  If so, can it be adapted to that design now?

Reply:
@ahsonkhan what is the status of having a public implementation of IBufferWriter<T>?